src/controllers/api/imageController.ts

```diff
@@ -1,25 +1,21 @@
 import { Request, Response, NextFunction } from 'express';
-import aws from 'aws-sdk';
 
-const s3 = new aws.S3({ params: { Bucket: 'dnd-5e-api-images' } });
+import { GetObjectCommand } from '@aws-sdk/client-s3';
+import { awsS3Client } from '../../util/index.js';
 
 const show = async (req: Request, res: Response, next: NextFunction) => {
   try {
     const params = {
       Bucket: 'dnd-5e-api-images',
       Key: req.url.slice(1),
-      Range: req.headers.range,
     };
-    return s3.getObject(params, function(err, data) {
-      if (err) {
-        res.status(200);
-        res.end('Error Fetching File');
-      } else {
-        res.writeHead(200, { 'Content-Type': data.ContentType });
-        res.write(data.Body, 'binary');
-        res.end(null, 'binary');
-      }
-    });
+
+    const s3Response = await awsS3Client.send(new GetObjectCommand(params));
+    const data = await s3Response.Body?.transformToByteArray();
+
+    res.writeHead(200, { 'Content-Type': 'image/png' });
+    res.write(data, 'binary');
+    res.end(null, 'binary');
```

Comment:
my async / await is a bit rusty, would love some 👀 on this

Reply:
I mean that looks fine. I think if you want, you could wrap them in a `try` block and add in:
```js
        res.status(200);
        res.end('Error Fetching File');
```

Reply:
added this try/catch